app/_src/policies/meshtimeout.md

```diff
@@ -0,0 +1,315 @@
+---
+title: Mesh Timeout (beta)
+---
+
+{% warning %}
+This policy uses new policy matching algorithm and is in beta state,
+it should not be mixed with [Timeout policy](../timeout).
+{% endwarning %}
+
+## TargetRef support matrix
+
+| TargetRef type    | top level | to  | from |
+| ----------------- | --------- | --- | ---- |
+| Mesh              | ✅         | ✅   | ✅    |
+| MeshSubset        | ✅         | ❌   | ❌    |
+| MeshService       | ✅         | ✅   | ❌    |
+| MeshServiceSubset | ✅         | ❌   | ❌    |
+| MeshGatewayRoute  | ❌         | ❌   | ❌    |
+
+To learn more about the information in this table, see the [matching docs](/docs/{{ page.version }}/policies/targetref).
+
+## Configuration
+
+This policy enables {{site.mesh_product_name}} to set timeouts on the inbound and outbound connections depending on the protocol. Using this policy you can configure TCP and HTTP timeouts. Timeout configuration is split into two sections: common configuration and http configuration. Common config is applied to both HTTP and TCP communication.
```

Comment:
I'd also drop a link somewhere that HTTP timeouts are only applied if a service is marked as HTTP https://kuma.io/docs/2.0.x/policies/protocol-support-in-kuma/#protocol-support-in-kuma

and _maybe_ that when gRPC is used, the HTTP timeout is appliedl

Reply:
about GRPC I was thinking about it when I wrote this documentation. We don't configure this timeouts for GRPC: https://github.com/kumahq/kuma/blob/master/pkg/plugins/policies/meshtimeout/plugin/xds/configurer.go#L59 Shouldn't be changed? So that we configure timeouts for GRPC connections?

Reply:
definitely, this should be changed!

Reply:
https://github.com/kumahq/kuma/pull/5841